### ISSUE TITLE
How to train the model using batch_size?

### ISSUE THREAD

**Issue by linbang** [apple2373/metapath2vec#8]
Looks like I need to modify “get_negative_samples” function in dataset.py. However,  i meet some bugs that cant be fixed...

**Comment by apple2373** [comment]
I don't remember much about my coding honestly, but I feel Tensorflow operation somehow forced me to use batch size of 1 because it does not support Heterogeneous negative sampling (equation 5 in the original paper). In the equation, we have to normalize with the same typed nodes only instead of all nodes, right? But when we use batch > 1 with multiple types, normalization constant is different for each batch. I think I could not do this with tf.nn.nce_loss, because they assume to normalize with the same constant for each element in the batch.  Maybe I was wrong or they already fixed it, but this is what I can remember now.

**Comment by linbang** [comment]
i have the same idea, thank you very much!
